### DISCUSSION TITLE
Sol2SPH keyword

### DISCUSSION THREAD

**Discussion by li2qn** [OpenRadioss/OpenRadioss#2225] [Q&A]
Hello everyone,

I am wondering if anyone has a simple example using the keyword Sol2SPH to convert solid tet4 elements to sph? I was looking around but cannot find any examples. Please help.

Regards
lee

**Comment by bertrand-altair** [comment]
Hi @li2qn ,
Here is a very basic example using tetra elements and SOL2SPH with a dummy failure criteria. I hope it can help.

[SOL2SPH_TETRA.zip](https://github.com/OpenRadioss/OpenRadioss/files/13922631/SOL2SPH_TETRA.zip)

Regards,

Bertrand,

**Reply by li2qn** [reply]
Thanks so much. @ Bertrand

**Reply by li2qn** [reply]
@ Bertrand. May I ask a question regrading the SPH features in Radioss? I had a simulation, all the SPH particles only subject to gravity load.  As a result, the particles should exit the opening of a funnel. However, in my simulation, the particles flew away and not exit the opening of the funnel at all.  Please see the attached image, could you please help me on this? Thanks so much

![image](https://github.com/OpenRadioss/OpenRadioss/assets/48695377/1e2e9730-16b5-4bb8-8761-b223a6d97700)

regards
Lee

**Reply by bertrand-altair** [reply]
Hi @li2qn,
It looks like the particles are able to cross the structure. How is the contact interface defined between SPH particles and the structure ? 
Regards,
Bertrand,

**Reply by li2qn** [reply]
Yes the particles are able to pass through the wall (rigid body) which is not expected. As a result of applying gravity load to all the particles, the particles should exit the opening of a funnel. However, in my simulation, the particles flew away and not exit the opening of the funnel at all. Please see attached contact interface for your reference. Thanks for my suggestion and comments.

![image](https://github.com/OpenRadioss/OpenRadioss/assets/48695377/87e38445-e8b3-4192-8424-fbf3b16f6dfb)

Regards
Lee

**Reply by bertrand-altair** [reply]
For contact between finite elements and SPH particles it is recommanded to use Igap=0 (constant gap) with Gapmin = size of particles (smoothing length divided by 2) because gap can not be computed for SPH particles.
Regards,
Bertrand,

**Reply by li2qn** [reply]
Thanks for let me know. I will try that. Do you know if I need to set the SPH/INOUT condition for the particles to exit from the opening? 

Regards
Lee

**Reply by bertrand-altair** [reply]
No there is no need to use an outlet unless you want to control pressure at the exit, but quality of the results will highly depend on the ratio between the size of particles and to the size of the opening.
Regards,
Bertrand,

**Reply by li2qn** [reply]
I was tested what you suggested too. Still have issues that particles flew everywhere. Any other idea?

**Reply by li2qn** [reply]
I am not able to choose the size of the particles right?

**Reply by bertrand-altair** [reply]
Hi @li2qn,
I had a look at the model and launched it only on a couple of cycles. The issue is that the particles are defined exactly on the surface of the finite element generating huge initial penetrations (and because of that huge added a mass -  DM/M = 1e+11). The best solution is to modify the position of SPH particles and put them inside the funel. Typically if you use HyperMesh to generate the particles you can define a wall clearance equal to half the diameter of the particles. You can also try to use Inacti=6 in /INTER/TYPE7 to treat initial penetrations. Using Inacti=6 the run starts properly but i am not 100% sure that it will prevent all particles crossing the surface of the funel.
Regards,
Bertrand,

**Reply by li2qn** [reply]
@Bertrand, Thanks for your investigation and explanation in details. That's really helped.  Do you know how can know/define the particle size? is there any cards in Radioss to do that?

**Reply by bertrand-altair** [reply]
No, you can not define it directly in RADIOSS or output it. It's coming from the particle discretisation.

**Reply by li2qn** [reply]
Thanks so much
